src/org/pahappa/systems/services/IncidentServiceImpl.java

```diff
@@ -41,7 +41,36 @@ public Incident saveIncident(Incident incident) throws Exception {
 
 	@Override
 	public Incident updateIncident(Incident incident) throws Exception {
-		// TODO Auto-generated method stub
+		//Check if both the comment and title sections have been filled
+		if (incident.getTitle()==null){
+			throw new ValidationFailedException("Please enter title");
+		}else if (incident.getTitle().isEmpty()){
+			throw new ValidationFailedException("Please enter title");
+		}else if (incident.getComment()==null){
+			throw new ValidationFailedException("Please enter comment");
+		}else if (incident.getComment().isEmpty()){
+			throw new ValidationFailedException("Please enter comment");
+		}
+
+		//Searching and updating the incident
+		Incident searchResult = findIncident(incident);
+		if (searchResult == null){
+			throw new ValidationFailedException("This record does not exist");
+		}else {
+			searchResult.setTitle(incident.getTitle());
+			searchResult.setComment(incident.getComment());
+			return searchResult;
+		}
+
+		//return null;
+	}
+
+	public Incident findIncident(Incident incident) throws Exception {
+		for(Incident item: incidents){
+			if(item.getTitle().contains(incident.getTitle())){
+				return  item;
```

Comment:
@ttcollins I don't think this function is working as expected. Since we didn't validate a function have a unique title. So you are likely to get many incidents with the same title.

Reply:
@reifred, the idea was to use the ID attribute, but users can't set their own IDs. So we figured it was ideal that they would not know the ID for the incident they would like to update since the incident also only returns a title.